### ISSUE TITLE
removeHs() should not remove H atoms that are contributing to the definition of a stereo bond

### ISSUE THREAD

**Issue by d-b-w** [rdkit/rdkit#1810]
I was looking at some imines with stereochemistry defined based on a hydrogen bonded to the nitrogen (e.g. F/C=N/[H]). It appears that RDkit does not recognize this hydrogen as being important to stereochemistry. It is removed by `rdkit.Chem.RemoveHs()`, and it is lost in ` rdkit.Chem.MolFromSmiles()`.

This is an interactive Python session demonstrating some of the issues relating to this. I've tried to call attention to the problem areas with assertions. This was also discussed in the RDKit-discuss mailing list: https://sourceforge.net/p/rdkit/mailman/message/36285575/

```
Python 3.6.2 (default, Jul 21 2017, 13:21:26) 
[GCC 4.9.3] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import rdkit
>>> print(rdkit.__version__)
2017.03.1
>>> from rdkit import Chem
>>> from rdkit.Chem import AllChem
>>> from rdkit.Chem import rdmolops
>>> cis_imine_smiles = '[H]/C=N/F'
>>> m = Chem.MolFromSmiles(cis_imine_smiles)
>>> Chem.MolToSmiles(m, isomericSmiles=True)
'N=CF'
>>> assert Chem.MolToSmiles(m, isomericSmiles=True) == cis_imine_smiles
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> has_h = next(Chem.SDMolSupplier('cis_imine.sdf', removeHs=False))
>>> Chem.MolToSmiles(has_h, isomericSmiles=True)
[H]/N=C(/[H])F
>>> no_h = rdmolops.RemoveHs(has_h)
>>> no_h.GetNumAtoms()
3
# this should be 4.
>>> Chem.MolToSmiles(no_h, isomericSmiles=True)
N=CF
>>> assert '/' in Chem.MolToSmiles(no_h, isomericSmiles=True):
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> has_h_again = rdmolops.AddHs(no_h)
>>> Chem.MolToSmiles(has_h_again, isomericSmiles=True)
[H]N=C([H])F
>>> AllChem.EmbedMolecule(has_h_again)
Fatal Python error: Segmentation fault

Current thread 0x00007faa949d8740 (most recent call first):
  File "<stdin>", line 1 in <module>
Segmentation fault
```

I've attached the sdf file gzipped:
[cis_imine.sdf.gz](https://github.com/rdkit/rdkit/files/1885345/cis_imine.sdf.gz)

**Comment by d-b-w** [comment]
Thanks!

**Comment by greglandrum** [comment]
@d-b-w : though I "fixed" this a couple of releases ago, #2422 has made me think about it a bit more and now I'm starting to wonder about things...

There's a good argument that the H atom in `F/C=N/[H]` should be kept: it's needed to set the stereochemistry at the double bond. 

However, another one of your examples is `[H]/C=N/F`. In this one there should be no stereochemistry at the double bond since C1 has two H atoms attached. So the stereochemistry cleanup code should really mark that bond as not having stereochem, which should lead to the H atom being removed. 

Does that make sense to you or is there a particular reason I'm not considering for why you raised that example?

**Comment by d-b-w** [comment]
whoop, no, `[H]/C=N/F` is complete nonsense. I don't know why I used that example. Maybe copy/paste error? It looks like the output that I was getting was `FC=N`, so I think that was meant to be another `F/C=N/[H]`.
